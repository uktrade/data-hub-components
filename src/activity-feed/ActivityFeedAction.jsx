import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Button from '@govuk-react/button'
import { SPACING, MEDIA_QUERIES } from '@govuk-react/constants'

const Link = styled('a')`
  margin-bottom: 0;
  margin-top: ${SPACING.SCALE_3};
  ${MEDIA_QUERIES.TABLET} {
    margin-left: ${SPACING.SCALE_3};
    margin-top: 0;
  }
`

const ActivityFeedAction = ({ text, link }) => (
  <Button
    key={text}
    as={Link}
    href={link}
    buttonColour="#dee0e2"
    buttonTextColour="#000"
  >
    {text}
  </Button>
)

ActivityFeedAction.propTypes = {
  text: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
}

export default ActivityFeedAction
