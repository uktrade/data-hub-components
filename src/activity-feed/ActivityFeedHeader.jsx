import React from 'react'
import PropTypes from 'prop-types'
import Button from '@govuk-react/button'
import { H2 } from '@govuk-react/heading'
import styled from 'styled-components'
import { SPACING, MEDIA_QUERIES } from '@govuk-react/constants'
import pluralize from 'pluralize'

const HeaderSummary = styled('div')`
  display: flex;
  flex-flow: row wrap;
  border-bottom: 2px solid #000;
  margin-bottom: ${SPACING.SCALE_2};
  padding-bottom: ${SPACING.SCALE_1};

  & > div {
    width: 100%;
    margin-bottom: ${SPACING.SCALE_1};

    ${MEDIA_QUERIES.TABLET} {
      width: 0;
      flex-grow: 1;
    }
  }
`

const HeaderCount = styled('div')`
  margin-top: ${SPACING.SCALE_1};

  & > h2 {
    font-weight: normal;
    font-size: 28px;
    margin-bottom: 0;
  }
`

const HeaderActions = styled('div')`
  text-align: right;

  & > button {
    margin-bottom: 0;
  }
`

const Link = styled.a`
  margin-bottom: 0;
`

export default class ActivityFeedHeader extends React.Component {
  static propTypes = {
    totalActivities: PropTypes.number,
    addContentText: PropTypes.string,
    addContentLink: PropTypes.string,
  }

  static defaultProps = {
    totalActivities: 0,
    addContentText: null,
    addContentLink: null,
  }

  render() {
    const { totalActivities, addContentText, addContentLink } = this.props
    const headerText = totalActivities
      ? pluralize('activity', totalActivities, true)
      : 'Activities'

    const showAddContentButton = addContentText && addContentLink

    return (
      <HeaderSummary>
        <HeaderCount>
          <H2>{headerText}</H2>
        </HeaderCount>
        <HeaderActions>
          {showAddContentButton && (
            <Button
              as={Link}
              href={addContentLink}
              buttonColour="#dee0e2"
              buttonTextColour="#000"
            >
              {addContentText}
            </Button>
          )}
        </HeaderActions>
      </HeaderSummary>
    )
  }
}
