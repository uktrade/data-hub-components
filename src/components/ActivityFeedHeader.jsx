import React from 'react'
import PropTypes from 'prop-types'

import ActivityFeedFilters from './ActivityFeedFilters'

export default class ActivityFeedHeader extends React.Component {
  static propTypes = {
    totalCards: PropTypes.number.isRequired,
  }

  render() {
    const { totalCards } = this.props

    return (
      <article>
        {totalCards} activities
        <button>Add activity</button>

        <ActivityFeedFilters />
      </article>
    )
  }
}
