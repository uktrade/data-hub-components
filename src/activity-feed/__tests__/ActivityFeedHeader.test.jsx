import React from 'react'
import renderer from 'react-test-renderer'

import ActivityFeedHeader from '../ActivityFeedHeader'
import ActivityFeedAction from '../ActivityFeedAction'

describe('ActivityFeedHeader', () => {
  test('renders header without props', () => {
    const tree = renderer.create(<ActivityFeedHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders header with all props defined', () => {
    const tree = renderer
      .create(
        <ActivityFeedHeader
          totalActivities={999}
          actions={
            <ActivityFeedAction
              text="Test button"
              link="http://testing.example.com"
            />
          }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders header with one activity', () => {
    const tree = renderer
      .create(<ActivityFeedHeader totalActivities={1} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
