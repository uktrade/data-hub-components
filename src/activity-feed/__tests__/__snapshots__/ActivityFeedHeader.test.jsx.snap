// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFeedHeader renders header with all props defined 1`] = `
.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 19px;
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  width: 100%;
  margin-top: 0;
  padding: 7px 10px;
  border: 2px solid transparent;
  border-radius: 0;
  color: #000;
  background-color: #dee0e2;
  box-shadow: 0 2px 0 #b5babe;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  -webkit-appearance: none;
  margin-bottom: 22px;
}

.c4:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

.c4:link,
.c4:visited,
.c4:active,
.c4:hover {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4::-moz-focus-inner {
  padding: 0;
  border: 0;
}

.c4:hover,
.c4:focus {
  background-color: #d0d3d6;
}

.c4:active {
  top: 2px;
  box-shadow: none;
}

.c4::before {
  content: "";
  display: block;
  position: absolute;
  top: -2px;
  right: -2px;
  bottom: -4px;
  left: -2px;
  background: transparent;
}

.c4:active::before {
  top: -4px;
}

.c4:disabled {
  opacity: 0.5;
  background: #dee0e2;
}

.c4:disabled:hover {
  background-color: #dee0e2;
  cursor: default;
}

.c4:disabled:focus {
  outline: none;
}

.c4:disabled:active {
  top: 0;
  box-shadow: 0 2px 0 #b5babe;
}

.c4 svg {
  max-width: 15px;
}

.c2 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.0416666666666667;
  display: block;
  margin-top: 0;
  margin-bottom: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  border-bottom: 2px solid #000;
  margin-bottom: 10px;
  padding-bottom: 5px;
}

.c0 > div {
  width: 100%;
  margin-bottom: 5px;
}

.c1 {
  margin-top: 5px;
}

.c1 > h2 {
  font-weight: normal;
  font-size: 28px;
  margin-bottom: 0;
}

.c3 {
  text-align: right;
}

.c3 > button {
  margin-bottom: 0;
}

.c5 {
  margin-bottom: 0;
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 19px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    width: auto;
  }
}

@media only screen and (min-width:641px) {
  .c4 svg {
    margin-left: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    margin-bottom: 32px;
  }
}

@media print {
  .c2 {
    color: #000;
  }
}

@media print {
  .c2 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c0 > div {
    width: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2"
      size="LARGE"
    >
      999 activities
    </h2>
  </div>
  <div
    className="c3"
  >
    <a
      className="c4 c5"
      disabled={false}
      href="http://testing.example.com"
    >
      Test button
    </a>
  </div>
</div>
`;

exports[`ActivityFeedHeader renders header with one activity 1`] = `
.c2 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.0416666666666667;
  display: block;
  margin-top: 0;
  margin-bottom: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  border-bottom: 2px solid #000;
  margin-bottom: 10px;
  padding-bottom: 5px;
}

.c0 > div {
  width: 100%;
  margin-bottom: 5px;
}

.c1 {
  margin-top: 5px;
}

.c1 > h2 {
  font-weight: normal;
  font-size: 28px;
  margin-bottom: 0;
}

.c3 {
  text-align: right;
}

.c3 > button {
  margin-bottom: 0;
}

@media print {
  .c2 {
    color: #000;
  }
}

@media print {
  .c2 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c0 > div {
    width: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2"
      size="LARGE"
    >
      1 activity
    </h2>
  </div>
  <div
    className="c3"
  />
</div>
`;

exports[`ActivityFeedHeader renders header without props 1`] = `
.c2 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.0416666666666667;
  display: block;
  margin-top: 0;
  margin-bottom: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  border-bottom: 2px solid #000;
  margin-bottom: 10px;
  padding-bottom: 5px;
}

.c0 > div {
  width: 100%;
  margin-bottom: 5px;
}

.c1 {
  margin-top: 5px;
}

.c1 > h2 {
  font-weight: normal;
  font-size: 28px;
  margin-bottom: 0;
}

.c3 {
  text-align: right;
}

.c3 > button {
  margin-bottom: 0;
}

@media print {
  .c2 {
    color: #000;
  }
}

@media print {
  .c2 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c0 > div {
    width: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2"
      size="LARGE"
    >
      Activities
    </h2>
  </div>
  <div
    className="c3"
  />
</div>
`;
